### ISSUE TITLE
Android CameraCalibration very laggy

### ISSUE THREAD

**Issue by Death111** [artoolkit/artoolkit5#131]
Hey,

when using the camera calibration app from the appstore and trying to calibrate, the app is very laggy. This is even worse with higher resolutions. Therefore it is not possible to get a good calibration. (Video: https://www.youtube.com/watch?v=ar3yfcKUF9E) This happens with my LG G4, as well as my Mi5 Pro.

This may relate to #125 as he also does not get a good calibration.

**Comment by the-max0r** [comment]
I can confirm this. It is impossible to generate a good calibration.

**Comment by ThorstenBux** [comment]
I can reproduce this with a Nexus 5 and high res a well. We are investigating into it.

**Comment by norips** [comment]
Hi everyone 
Does the sources of the calibration app are available ? 
Thanks

**Comment by Plantaer** [comment]
I'm confirming too. It's been a year and nothing changed?

**Comment by philip-lamb** [comment]
Calibration is computationally intensive. With HD frames on a mobile device, identifying the chessboard corners can take several seconds per frame. The best option is mount the device on a tripod.

**Comment by Death111** [comment]
It felt like, if it does not detect all 'corners' that it will lag. Then, when all corners are in the frame it was smooth. Not sure how this comes. I guess when it detected all corners once, the algorithm will use a optimized search algorithm (only search in small areas as before)? It somehow felt weird and I thought that a timeout may help with this problem: If after 1 second the corners are not found, drop the frame. I don't know if this makes sense, as I do not know the implementation ;)

**Comment by the-max0r** [comment]
Wouldn't it be possible to do the corner detection "offline", e.g. take a bunch of photos and after that, find the corners and do the calibration?
Another thing I want to remind is prevention of uploading a bad calibration dataset (see https://github.com/artoolkit/artoolkit5/issues/125). After the app uploaded a bad calibration dataset, my Nexus 9 was unusable, because it always loaded this bad dataset.

**Comment by Plantaer** [comment]
I agree. My Sony Z1 is also unusable. Bad calibration data should be overwritten by a new one.
